src/commons/crypto/signing/signers/pkcs11/signer.rs

```diff
@@ -0,0 +1,89 @@
+use pkcs11::types::{CKO_PRIVATE_KEY, CKO_PUBLIC_KEY};
+use rpki::repository::crypto::{
+    signer::KeyError, KeyIdentifier, PublicKey, PublicKeyFormat, Signature, SignatureAlgorithm, SigningError,
+};
+
+use crate::commons::crypto::{signers::pkcs11::Pkcs11Signer, SignerError};
+
+// Implement the functions defined by the `Signer` trait because `SignerProvider` expects to invoke them, but as the
+// dispatching is not trait based we don't actually have to implement the `Signer` trait.
+impl Pkcs11Signer {
```

Comment:
It's a bit confusing to me to see two files `internal.rs` and `siging.rs` with this implementation being here. Why not just one file? My guess is that `internal` was supposed to be hidden? But looking at `mod.rs` it's all pub.

Reply:
This was indeed because the `internal.rs` functions were meant to be used "internally" within the signing part of Krill, because I felt that `internal.rs` was already quite long and because the functions implemented for the `Signer` trait as the public interface made a nice chunk to separate out. However the `Signer` trait is no longer implemented thanks to your earlier review feedback, and the `pub(crate|super|etc)` attributions on most functions and structs probably needs some cleaning up.

Reply:
I'll move the functions back into `internal.rs` for now and remove `signer.rs`. It might warrant some refactoring later.

Reply:
Done.